Marlin/src/gcode/feature/ft_motion/M493.cpp

```diff
@@ -28,38 +28,48 @@
 #include "../../../module/ft_motion.h"
 #include "../../../module/stepper.h"
 
+void say_shaper_type(const ftMotionCmpnstr_t t) {
+  SERIAL_ECHOPGM(" axis ");
+  switch (t) {
+    default: break;
+    case ftMotionCmpnstr_ZV:    SERIAL_ECHOPGM("ZV");        break;
+    case ftMotionCmpnstr_ZVD:   SERIAL_ECHOPGM("ZVD");       break;
+    case ftMotionCmpnstr_ZVDD:  SERIAL_ECHOPGM("ZVDD");      break;
+    case ftMotionCmpnstr_ZVDDD: SERIAL_ECHOPGM("ZVDDD");     break;
+    case ftMotionCmpnstr_EI:    SERIAL_ECHOPGM("EI");        break;
+    case ftMotionCmpnstr_2HEI:  SERIAL_ECHOPGM("2 Hump EI"); break;
+    case ftMotionCmpnstr_3HEI:  SERIAL_ECHOPGM("3 Hump EI"); break;
+    case ftMotionCmpnstr_MZV:   SERIAL_ECHOPGM("MZV");       break;
+  }
+  SERIAL_ECHOPGM(" shaping");
+}
+
 void say_shaping() {
   // FT Enabled
   SERIAL_ECHO_TERNARY(ftMotion.cfg.mode, "Fixed-Time Motion ", "en", "dis", "abled");
 
   // FT Shaping
   #if HAS_X_AXIS
-    if (ftMotion.cfg.mode > ftMotionMode_ENABLED) {
-      SERIAL_ECHOPGM(" with ");
-      switch (ftMotion.cfg.mode) {
-        default: break;
-        case ftMotionMode_ZV:    SERIAL_ECHOPGM("ZV");        break;
-        case ftMotionMode_ZVD:   SERIAL_ECHOPGM("ZVD");       break;
-        case ftMotionMode_ZVDD:  SERIAL_ECHOPGM("ZVDD");      break;
-        case ftMotionMode_ZVDDD: SERIAL_ECHOPGM("ZVDDD");     break;
-        case ftMotionMode_EI:    SERIAL_ECHOPGM("EI");        break;
-        case ftMotionMode_2HEI:  SERIAL_ECHOPGM("2 Hump EI"); break;
-        case ftMotionMode_3HEI:  SERIAL_ECHOPGM("3 Hump EI"); break;
-        case ftMotionMode_MZV:   SERIAL_ECHOPGM("MZV");       break;
-        //case ftMotionMode_DISCTF: SERIAL_ECHOPGM("discrete transfer functions"); break;
-        //case ftMotionMode_ULENDO_FBS: SERIAL_ECHOPGM("Ulendo FBS."); return;
-      }
-      SERIAL_ECHOPGM(" shaping");
+    if (CMPNSTR_HAS_SHAPER(X_AXIS)) {
+      SERIAL_ECHOPGM(" with X/A");
+      say_shaper_type(ftMotion.cfg.cmpnstr[X_AXIS]);
+    }
+  #endif
+  #if HAS_Y_AXIS
+    if (CMPNSTR_HAS_SHAPER(Y_AXIS)) {
+      SERIAL_ECHOPGM(" and with Y/B");
```

Comment:
We can say "with X shaper" / "with Y shaper" on standard Cartesian, and just print "with A shaper" / "with B shaper" on Core XY. That should make it a little clearer for both types of setups.